community/kernel/src/main/java/org/neo4j/kernel/impl/recovery/StoreRecoverer.java

```diff
@@ -71,21 +75,24 @@ public boolean recoveryNeededAt( File dataDir, long currentLogVersion )
         }
 
         PhysicalLogFiles logFiles = new PhysicalLogFiles( dataDir, fs );
-        File log = logFiles.getLogFileForVersion( currentLogVersion );
 
-        if ( !fs.fileExists( log ) )
-        {
-            // This most likely means that the db has been cleanly shut down, i.e. force then inc log version,
-            // then NOT creating a new log file (will be done the next startup)
-            return false;
-        }
-        try ( StoreChannel logChannel = fs.open( log, "r" ) )
+        LogEntryReader<ReadableVersionableLogChannel> reader = new LogEntryReaderFactory().versionable();
+
+        LatestCheckPointFinder finder = new LatestCheckPointFinder( logFiles, fs, reader );
+
+        CheckPoint checkPoint = finder.find( currentLogVersion );
+        if ( checkPoint == null )
         {
-            return LogRecoveryCheck.recoveryRequired(logChannel);
+            return true;
         }
+
+        NeoStoreUtil neoStoreUtil = new NeoStoreUtil( dataDir, fs );
+        LogPosition lastTxCommittedLogPosition = new LogPosition(
+                neoStoreUtil.getLastCommittedTxLogVersion(), neoStoreUtil.getLastCommittedTxLogByteOffset() );
+        return !checkPoint.getLogPosition().equals( lastTxCommittedLogPosition );
```

Comment:
To reiterate: I don't see a need for storing the log position in the checkpoint record. If we just had the transaction id we could compare instead of getting the last log position, which feels a bit awkward.